Describe how the Knative channel name is constructed (#222)
kyma-project/community: sig-and-wg/wg-knative/knative-eventing-adoption.md

>Note: A Kyma Event type is just a logical concept. The actual Event type is a Fully Qualified Event Type composed of `application(source id) + event type + event type version`.

## Knative channel naming convention

>  The Knative channel name follows this pattern: `{event-source-id}-{event-type}-{event-type-version}`. In order to avoid naming conflicts, use the following guidelines when creating the channel name: 

Reviewer: ```suggestion
 The Knative channel naming follows this pattern: `{event-source-id}-{event-type}-{event-type-version}`. In order to avoid naming conflicts, the following rules are considered when creating the channel name: 
```

 1. If the event component's name already contains a hyphen, a second hyphen is added. For example, `external-application` becomes `external--application`.

>  2. If the event component's name contains a period, the period is replaced with `-dot-`. For example, `order.created` becomes `order-dot-created`.

Reviewer: ```suggestion
 2. If the event component's name already contains a period, the period is replaced with `-dot-`. For example, `order.created` becomes `order-dot-created`.
```

>Note: A Kyma Event type is just a logical concept. The actual Event type is a Fully Qualified Event Type composed of `application(source id) + event type + event type version`.

## Knative channel naming convention
 The Knative channel name is composed of three components: the event **source-id**, **event-type** and the **event-type-version** joined by the **hyphen** character. In order to avoid channel naming collisions, the Knative channel name is constructed in the following order:
 1. In case any of the event components contains the hyphen character, then it will be escaped by another hyphen character (e.g. external-application will be external--application).
 2. In case any of the event components contains the period character, then it will be replaced by the **-dot-** character sequence (e.g. order.created will be order-dot-created).
 3. After applying step 1 and 2 for each component of the three event components, the three components output will be joined by the hyphen character (e.g. the channel name **external--application-order-dot-created-v1** is construcled from the source-id **external-application**, event-type **order.created** and the event-type-version **v1**).
 

>  Here are some example event components that can cause channel naming collisions:

Reviewer: ```suggestion
The table includes sample channel names created according to the naming pattern:
```
